src/k8s-engine/lima.ts

```diff
@@ -1822,10 +1820,15 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
 
   protected async getHostIPAddr(): Promise<string> {
     try {
-      const lines = (await this.limaWithCapture('shell', '--workdir=.', MACHINE_NAME, 'ip', 'route', 'list', 'eth0')).split(/\n/);
-      const fields = lines[0].split(/\s+/);
+      while (true) {
+        const stdout = await this.limaWithCapture('shell', '--workdir=.', MACHINE_NAME, 'ip', 'route', 'list', 'eth0');
+        const line = stdout.split(/\n/).find(line => /\bvia .* dev eth0\b/.test(line));
+        const match = /\bvia (.*) dev eth0\b/.exec(line ?? '');
 
-      return fields[2];
+        if (match) {
+          return match[1];
+        }
+      }
```

Comment:
I think you need a `sleep` at this point, possibly doing exponential backoff, and be prepared if no IPAddr is detected after some length of time, possibly 3-5 minutes. Otherwise there's too much spinning going on here (potentially).

Reply:
Exponential backoff added in 2d03778b897277063131943bd76205e079dfaf03